### ISSUE TITLE
Capture scheduler error messages and print to user

### ISSUE THREAD

**Issue by justinGilmer** [glotzerlab/signac-flow#103]
### Feature description

When the user runs `submit` for a system with a scheduler management system, any error messages from the scheduler command itself should be captured and displayed to the user.

### Proposed solution

If `subprocess.py` can capture the STDOUT and STDERR of the command it calls, and returns that to the user when calling `submit` that would drastically improve the debugging experience. 

### Additional context

Just today I was submitting some jobs to a `torque` system, however I did not realize that I had over-allocated the maximum hours I could allocate. 

The error message mentioned that there was an error with `qsub` and the temporary file that was the batch script. It also mentioned that it returned error code 255, unfortunately that is a generic error code. If `subprocess` can capture the STDOUT and STDERR of `qsub` or the slurm equivalent, that would be excellent in debugging.

**Comment by vyasr** [comment]
I agree that this would be a good idea. I'll try to have a look at it.

**Comment by justinGilmer** [comment]
Here is the selection of code from [torque.py](https://github.com/glotzerlab/signac-flow/blob/fc7916fae3d9e10bf9828a4b296ce93321e86e61/flow/scheduling/torque.py#L141) where I think the check should be done
```python3
        if pretend:
            print("# Submit command: {}".format(' '.join(submit_cmd)))
            print(script)
            print()
        else:
            with tempfile.NamedTemporaryFile() as tmp_submit_script:
                tmp_submit_script.write(str(script).encode('utf-8'))
                tmp_submit_script.flush()
                output = subprocess.check_output(
                    submit_cmd + [tmp_submit_script.name])
            jobsid = output.decode('utf-8').strip()
return jobsid
```

What I think should be changed:
```python3
        if pretend:
            print("# Submit command: {}".format(' '.join(submit_cmd)))
            print(script)
            print()
        else:
            with tempfile.NamedTemporaryFile() as tmp_submit_script:
                tmp_submit_script.write(str(script).encode('utf-8'))
                tmp_submit_script.flush()
				try:
                	output = subprocess.check_output(
                    	submit_cmd + [tmp_submit_script.name])
					jobsid = output.decode('utf-8').strip()
				except subprocess.CalledProcessError as e:
					print(e.output)
return jobsid
```
All that I changed was wrapping the `subprocesss` call in a try,except block.

I think all that is required is capturing the exception raised and then printing that out. However, I feel like this will be a lot of code replicating for each scheduler file.

**Comment by csadorf** [comment]
Resolved with merge of #104 .
